packages/fxa-settings/src/components/Settings/index.tsx

```diff
@@ -60,13 +60,65 @@ export const Settings = ({
   ]);
 
   useEffect(() => {
+    /**
+     * If we have an active session, we need to handle the possibility
+     * that it will reflect the session for the current tab. It's
+     * important to note that there is also a cache in local storage, and
+     * as such it is shared between all tabs. So, in the event a user has
+     * account A signed in on tab 1, and account B signed in on tab 2, local
+     * storage will reflect the account uid of whichever account was the last
+     * to be sign in.
+     *
+     * By noting the window focus, we actively swap the current account uid
+     * in local storage so that it matches the apollo cache's account uid,
+     * which is held in page memory, thereby fixing this discrepancy.
+     *
+     * Having multiple things cached in multiple places is never great, so we
+     * have a ticket in the backlog for cleaning this up, and avoiding this
+     * hack in the future. See FXA-9875 for more info.
+     */
     function handleWindowFocus() {
-      if (account.uid !== currentAccount()?.uid) {
-        setCurrentAccount(account.uid);
-      }
-      if (session.isDestroyed) {
+      // Try to retrieve the active account uid from the apollo cache.
+      const accountUidFromApolloCache = (() => {
+        try {
+          return account.uid;
+        } catch {}
+        return undefined;
+      })();
+
+      // During normal usage, we should not see this. However, if this happens many
+      // functions on the page would be broken, because it indicates the apollo
```

Comment:
nit: I think the word 'cache' is missing from the end of this line.  (but no need to update unless you are going to make other changes; no need to trigger CI just for it.)